### PR TITLE
Network requests

### LINKED ISSUE #14: Handle server error
When the server returns a 500 error, the user will see proper error handling

### LINKED ISSUE #15: Write apiCall functions
As a dev I need to write two apiCall functions in their own apiCall file:
1. fetch data for all movies
2. fetch data for a specific movie

### LINKED ISSUE #16: Test api fetch calls
As a dev I need to write tests for my api fetch calls that will mock out the api calls

### LINKED ISSUE #17: Refactor Movie component
As a dev I need to refactor the movie component so that it will reflect the data from the Movie fetch call

### LINKED ISSUE #18: Refactor App component
As a dev, I need to refactor the App component so that the data in this.state comes from an api and not from a linked file.

### PR DESCRIPTION
**Pull request by baileydunning**
### What does this PR do?
+ Set up API calls to have app run off fetched data for all movies and an individual movie
+ Write tests to ensure API calls are being called correctly
+ Update conditional rendering to reflect when loading or when there is an error

### What files were changed?
+ `App.js`
+ `App.test.js`
+ `Movie.js`
+ `Movie.test.js`

### Issues:
+ Closes #14
+ Closes #15 
+ Closes #16 
+ Closes #17 
+ Closes #18 

### Next Steps:
+ Fix CSS styling (especially for Movie component)

### CODE REVIEW

File: src/App/App.js [hunk 1]
```diff
@@ -27,29 +34,30 @@ class App extends Component {
...
-        return movie.id === id
-      })
-    )
+    getMovieData(id)
+    .then(data => this.setState({ selectedMovie: data.movie }))
+    .catch(error => this.setState({ error: error.message }))
   }
 
   render() {
     return (
```
Comment (bruce-gordon) [inline src/App/App.js:43]: I like the way we wrote this conditional rendering.

File: src/App/App.test.js [hunk 2]
```diff
@@ -1,17 +1,59 @@
...
+          "Action"
+        ],
+        "budget": 0,
+        "revenue": 0,
+        "runtime": 82,
+        "tagline": "",
+        "average_rating": 6.666666666666667
+      }
+    })
+    fireEvent.click(screen.getByAltText('movie-poster'))
```
Comment (bruce-gordon) [inline src/App/App.test.js:55]: This worked out really well.  I'm happy with how we approached this.

File: src/Movie/Movie.js [hunk 3]
```diff
@@ -2,14 +2,33 @@ import React from 'react'
...
-          <h2>{movie.title}</h2>
-          <p><b>Average Rating:</b> {movie.average_rating.toFixed(1)}</p>
-          <p><b>Release Date:</b> {new Date(movie.release_date).toDateString()}</p>
+          <h2>{ movie.title }</h2>
+          <h3>{ movie.tagline }</h3>
+          <p>{ movie.overview }</p>
+          <p><b>Average Rating:</b> { movie.average_rating.toFixed(1) }</p>
+          <p><b>Release Date:</b> { new Date(movie.release_date).toDateString() }</p>
+          <p><b>Runtime:</b> { movie.runtime } minutes</p>
+          { convertNumber(movie.budget, 'Budget') }
```
Comment (bruce-gordon) [inline src/Movie/Movie.js:28]: I'm so glad we took the time to experiment with this.  I think it makes this dryer and more dynamic.

**Review by bruce-gordon — APPROVED** [review]

File: src/App/App.js [hunk 4]
```diff
@@ -2,18 +2,25 @@ import React, { Component } from 'react';
...
 import '../App/App.scss';
 
 class App extends Component {
   constructor() {
     super()
     this.state = {
-      movies: movieData,
-      selectedMovie: null
+      movies: [],
+      selectedMovie: null,
```
Comment (letakeane) [inline src/App/App.js:13]: Nice job setting initial values for movies and error! I would consider making selectedMovie an empty object.

File: src/App/App.js [hunk 5]
```diff
@@ -2,18 +2,25 @@ import React, { Component } from 'react';
...
+      error: ''
     }
   }
 
+  componentDidMount = () => {
```
Comment (letakeane) [inline src/App/App.js:18]: Great use of the lifecycle method!

File: src/App/App.test.js [hunk 6]
```diff
@@ -1,17 +1,59 @@
...
-import Movie from '../Movie/Movie.js'
-import Header from '../Header/Header.js'
-import ThumbnailContainer from '../Thumbnail-Container/ThumbnailContainer.js'
+import userEvent from '@testing-library/user-event'
+import { getAllMovies, getMovieData } from '../apiCalls.js'
+jest.mock('../apiCalls.js')
+
 
 describe('App', () => {
   beforeEach(() => {
```
Comment (letakeane) [inline src/App/App.test.js:11]: This is a great use of a beforeEach!

File: src/Movie/Movie.js [hunk 7]
```diff
@@ -2,14 +2,33 @@ import React from 'react'
...
   return (
     <section className='movie-section' style={{ background: `url(${ movie.backdrop_path })`}}>
       <article className="movie-info">
         <img src={ movie.poster_path } alt="movie-poster" className="movie-poster"></img>
         <div className="movie-text">
-          <h2>{movie.title}</h2>
-          <p><b>Average Rating:</b> {movie.average_rating.toFixed(1)}</p>
-          <p><b>Release Date:</b> {new Date(movie.release_date).toDateString()}</p>
+          <h2>{ movie.title }</h2>
+          <h3>{ movie.tagline }</h3>
```
Comment (letakeane) [inline src/Movie/Movie.js:23]: Some movies don't have taglines - how are you handling this? Perhaps some conditional rendering would be good!

File: src/Movie/Movie.js [hunk 8]
```diff
@@ -2,14 +2,33 @@ import React from 'react'
...
+          <p>{ movie.overview }</p>
+          <p><b>Average Rating:</b> { movie.average_rating.toFixed(1) }</p>
+          <p><b>Release Date:</b> { new Date(movie.release_date).toDateString() }</p>
+          <p><b>Runtime:</b> { movie.runtime } minutes</p>
+          { convertNumber(movie.budget, 'Budget') }
+          { convertNumber(movie.revenue, 'Revenue') }
+          <p><b>Genres:</b></p>
+          <ul>{genres}</ul>
```
Comment (letakeane) [inline src/Movie/Movie.js:31]: Love this bit of functionality!

**Pull request merged**